department-of-veterans-affairs/caseflow
# CASEFLOW-948 CAVC entry form: JMR/JMPR dynamic issue validation

Resolves [CASEFLOW-948](https://vajira.max.gov/browse/CASEFLOW-948)

As a CAVC Litigation Support user, when I select/de-select issues, I want an informational banner noting all issues need to be selected for JMR Remands and at least one needs to be selected for JMPR, so that I do not have to hit submit before I see an error. 

### Description
For the CAVC entry form when JMR or JMPR is selected, show informational banner based on number of issues selected.

### Acceptance Criteria

Scenario: Given I am a CAVC Litigation Support user entering a CAVC Remand
When I identify the remand type is JMR and I de-select one or more issues
- [x] Then an informational banner displays below the issues section with the following copy: "For a JMR you must select all issues to proceed"

When I identify the remand type is a JMPR **or MDR** and I de-select all issues
- [x] Then an informational banner displays below the issues section with the following copy: "For a JMPR you must select at least one issue to proceed" 

(The MDR requirement was added during a demo to Ann-Marie.)

### Testing Plan
1. Open a Rails console and create a ready for CAVC appeal, note the `stream_docket_number` so you can search for the appeal in the UI or the `uuid`.
```ruby
original_appeal = FactoryBot.create(:appeal, :dispatched, associated_judge: JudgeTeam.first.judge, associated_attorney: JudgeTeam.first.attorneys.first);
FactoryBot.create_list(:request_issue, 2, :rating, :with_rating_decision_issue, decision_review: original_appeal, veteran_participant_id: Veteran.first.participant_id, contested_issue_description: "issue_description", notes: "issue_notes");
original_appeal.uuid
```
2. Login as a CAVC lit support user 
3. Search for the dispatched appeal you just created and select it to go to Case Details: `http://localhost:3000/queue/appeals/<UUID>`
4. Click the "+ Add CAVC Remand" button in the "Post-Dispatch actions" section
5. Fill out the "Add a Court Remand" form
   * select JMR: Unselect an issue and notice the informational banner.
   * select JMPR: Unselect all issue and notice the informational banner.
   * select MDR: Unselect all issue and notice the informational banner.
6. Play around, stress test, and find bugs early

### User Facing Changes
 - [ ] Screenshots of UI changes

 BEFORE|AFTER
 ---|---
![image](https://user-images.githubusercontent.com/55255674/109034764-965bd880-768d-11eb-9422-23a04f46dd08.png) | ![image](https://user-images.githubusercontent.com/55255674/109034054-dc646c80-768c-11eb-958e-18f6d98826b1.png)
![image](https://user-images.githubusercontent.com/55255674/109034815-a2479a80-768d-11eb-8516-3aeedae9074d.png) | ![image](https://user-images.githubusercontent.com/55255674/109034096-e8502e80-768c-11eb-8baf-ec045f604adb.png)


### Storybook Story
* [ ] ~Write a separate story (within the same file) for each discrete variation of the component~ This is not a sufficiently controlled component, so I can change the remand subtype or select issues.
